src/pages/home/report/ReportActionCompose.js

```diff
@@ -446,6 +441,56 @@ class ReportActionCompose extends React.Component {
         ];
     }
 
+    /**
+     * Build the suggestions for mentions
+     * @param {Object} personalDetails
+     * @param {String} [searchValue]
+     * @returns {Object}
+     */
+    getMentionOptions(personalDetails, searchValue = '') {
+        const suggestions = [];
+
+        if (CONST.MENTION_SUGGESTER.HERE_TEXT.includes(searchValue)) {
+            suggestions.push({
+                text: CONST.MENTION_SUGGESTER.HERE_TEXT,
+                alternateText: this.props.translate('mentionSuggestions.hereAlternateText'),
+                icons: [
+                    {
+                        source: Expensicons.Megaphone,
+                        type: 'avatar',
+                    },
+                ],
+            });
+        }
+
+        const searchRegex = new RegExp(searchValue, 'i');
+
+        const filteredPersonalDetails = _.filter(_.values(personalDetails), (detail) => {
+            if (searchValue && !searchRegex.test(`${detail.displayName} ${detail.login}`.trim())) {
+                return false;
+            }
+            return true;
+        });
+
+        const sortedPersonalDetails = _.sortBy(filteredPersonalDetails, (detail) => detail.displayName || detail.login);
```

Comment:
You could use `_.first(sortedPersonalDetails, 5)` to get the first 5 of these, then do the `_.each`. That way you wouldn't be wasting time pushing suggestions that are just going to get sliced. I think that would resolve @eVoloshchak's concern.